wallet/controllers_v3.go

```diff
@@ -671,3 +671,75 @@ func GetLinkingInfoV3(s *Service) func(w http.ResponseWriter, r *http.Request) *
 		return handlers.RenderContent(ctx, info, w, http.StatusOK)
 	}
 }
+
+// DisconnectCustodianLinkV3 - produces an http handler for the service s which handles disconnect
+// state for a deposit account linking
+func DisconnectCustodianLinkV3(s *Service) func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+	return func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var (
+			ctx       = r.Context()
+			id        = new(inputs.ID)
+			custodian = new(CustodianName)
+		)
+		// get logger from context
+		logger, err := appctx.GetLogger(ctx)
+		if err != nil {
+			// no logger, setup
+			ctx, logger = logging.SetupLogger(ctx)
+		}
+
+		// get payment id
+		if err := inputs.DecodeAndValidateString(ctx, id, chi.URLParam(r, "paymentID")); err != nil {
+			logger.Warn().Str("paymentID", err.Error()).Msg("failed to decode and validate paymentID from url")
+			return handlers.ValidationError(
+				"error validating paymentID url parameter",
+				map[string]interface{}{
+					"paymentID": err.Error(),
+				},
+			)
+		}
+
+		// get custodian name
+		if err := inputs.DecodeAndValidateString(ctx, custodian, chi.URLParam(r, "custodian")); err != nil {
```

Comment:
interesting. this interface copies to byte slice and then back to a string.

Reply:
Trying to make decode and validate as generic as possible, taking in a byte slice.